source/internal/interactive_scene.cpp

```diff
@@ -24,11 +24,27 @@ int cache_scenes(interactive_session_internal& session)
 	scenesArray.CopyFrom(replyScenesArray, session.scenesRoot.GetAllocator());
 	session.scenesRoot.AddMember(RPC_PARAM_SCENES, scenesArray, session.scenesRoot.GetAllocator());
 
+	RETURN_IF_FAILED(update_control_pointers(session));
+
+	return MIXER_OK;
+}
+
+int update_control_pointers(interactive_session_internal& session, const char* sceneId)
+{
 	// Iterate through each scene and set up a pointer to each control.
 	int sceneIndex = 0;
 	for (auto& scene : session.scenesRoot[RPC_PARAM_SCENES].GetArray())
 	{
 		std::string scenePointer = "/" + std::string(RPC_PARAM_SCENES) + "/" + std::to_string(sceneIndex++);
+		auto thisSceneId = scene[RPC_SCENE_ID].GetString();
+		if (nullptr != sceneId)
```

Comment:
Is this nullptr check because SCENE_ID wouldn't exist for some reason? If that's at all possible you should actually check if the property exists first using `HasMember`.

Reply:
Also with regards to the `HasMember` check it depends how defensive we want to be - in other places in the code we're trusting these fields to be there (and according to the current protocol they should) so I don't check that the member exists here because this value only gets updated with an object from the socket